osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -153,10 +163,55 @@ protected virtual RulesetKeyBindingContainer CreateKeyBindingContainer(RulesetIn
 
         public class RulesetKeyBindingContainer : DatabasedKeyBindingContainer<T>
         {
+            public event Func<T?, List<T>, bool> HandleBindings;
```

Comment:
* This event signature is confusing. There is no reason to split between the single action nullable case and the list-of-all-actions collection case.
* By convention events are usually named `OnXYZ`. In this case I'd probably suggest `OnRulesetAction`.